pkg/products/marin3r/reconciler.go

```diff
@@ -140,8 +148,25 @@ func (r *Reconciler) Reconcile(ctx context.Context, installation *integreatlyv1a
 		return phase, err
 	}
 
-	phase, err = NewRateLimitServiceReconciler(productNamespace, externalRedisSecretName).
-		ReconcileRateLimitService(ctx, client)
+	phase, err = r.reconcilePromStatsdExporter(ctx, client, productNamespace)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		events.HandleError(r.recorder, installation, phase, fmt.Sprintf("Failed to reconcile Prometheus StatsD exporter cr"), err)
+		return phase, err
+	}
+	statsdConfig := StatsdConfig{
+		Host: statsdHost,
+		Port: statsdPort,
+	}
+
+	phase, err = r.reconcilePromStatsdExporterService(ctx, client, productNamespace)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		events.HandleError(r.recorder, installation, phase, fmt.Sprintf("Failed to reconcile Prometheus StatsD exporter service"), err)
+		return phase, err
+	}
+
+	rlReconciler := NewRateLimitServiceReconciler(productNamespace, externalRedisSecretName)
+	rlReconciler.WithStatsdConfig(statsdConfig)
+	phase, err = rlReconciler.ReconcileRateLimitService(ctx, client)
```

Comment:
```suggestion
	phase, err = NewRateLimitServiceReconciler(productNamespace, externalRedisSecretName).
	    WithStatsdConfig(statsdConfig).
	    ReconcileRateLimitService(ctx, client)
```

Just a tiny note. The `WithStatsdConfig` function returns the `RateLimitServiceReconciler` instance, so you can instantiate it, add the config, and reconcile it in the same statement, without allocating a variable for it

Reply:
Yep, its cleaner